osmoutils/cache_ctx.go

```diff
@@ -30,3 +32,23 @@ func ApplyFuncIfNoError(ctx sdk.Context, f func(ctx sdk.Context) error) (err err
 	}
 	return err
 }
+
+// This function error logs the recoveryError, along with the stacktrace, if it can be parsed.
+// If not emits them to stdout
+func PrintPanicRecoveryError(ctx sdk.Context, recoveryError interface{}) {
+	errStackTrace := string(debug.Stack())
+	switch e := recoveryError.(type) {
+	case string:
+		ctx.Logger().Error("Recovering from panicrecovered (string) panic: " + e)
```

Comment:
Not sure what "panicrecovered" means here
```suggestion
		ctx.Logger().Error("Recovering from (string) panic: " + e)
```

Reply:
good catch, it was copy/paste error!